--- boost/token_iterator.hpp	2021-06-22 23:38:41.835565527 +0900
+++ boost/token_iterator_new.hpp	2021-06-22 23:38:16.195567055 +0900
@@ -38,8 +38,9 @@
           , const Type&
         >
   {
-
-#ifdef __DCC__ 
+#if defined(__NEC__)
+      friend class ::boost::iterators::iterator_core_access;
+#elif  defined(__DCC__)
       friend class boost::iterator_core_access; 
 #else 
       friend class iterator_core_access; 
--- boost/algorithm/string/find_iterator.hpp	2021-03-10 08:23:10.000000000 +0900
+++ boost/algorithm/string/find_iterator_new.hpp	2021-06-23 00:40:26.658344702 +0900
@@ -56,7 +56,11 @@
         {
         private:
             // facade support
+#if defined(__NEC__)
+            friend class ::boost::iterators::iterator_core_access;
+#else 
             friend class ::boost::iterator_core_access;
+#endif  
 
         private:
         // typedefs
@@ -212,7 +216,11 @@
         {
         private:
             // facade support
+#if defined(__NEC__)
+            friend class ::boost::iterators::iterator_core_access;
+#else 
             friend class ::boost::iterator_core_access;
+#endif  
 
         private:
         // typedefs
--- libs/thread/src/pthread/thread.cpp	2021-03-10 08:23:11.000000000 +0900
+++ libs/thread/src/pthread/thread_new.cpp	2021-06-27 09:55:03.091761906 +0900
@@ -497,7 +497,7 @@
         return (count>0)?count:0;
 #elif defined(__VXWORKS__)
         cpuset_t set =  ::vxCpuEnabledGet();
-  #ifdef __DCC__
+  #if defined(__DCC__) || defined(__NEC__)
         int i;
         for( i = 0; set; ++i)
         {
--- boost/interprocess/detail/os_thread_functions.hpp	2021-03-10 08:23:10.000000000 +0900
+++ boost/interprocess/detail/os_thread_functions_new.hpp	2021-06-27 09:53:59.291765708 +0900
@@ -485,7 +485,7 @@
       }
    #elif defined(__VXWORKS__)
       cpuset_t set =  ::vxCpuEnabledGet();
-    #ifdef __DCC__
+    #if defined(__DCC__) || defined(__NEC__)
       int i;
       for( i = 0; set; ++i)
           {
--- boost/system/detail/generic_category_message.hpp	2021-07-01 01:42:29.070924096 +0900
+++ boost/system/detail/generic_category_message_new.hpp	2021-07-01 01:42:21.421924552 +0900
@@ -23,7 +23,7 @@
 namespace detail
 {
 
-#if defined(__GLIBC__)
+#if defined(__GLIBC__) && !defined(__NEC__)
 
 // glibc has two incompatible strerror_r definitions
 
